rust/lance-core/src/datatypes/field.rs

```diff
@@ -980,16 +992,42 @@ impl TryFrom<&ArrowField> for Field {
     type Error = Error;
 
     fn try_from(field: &ArrowField) -> Result<Self> {
+        let mut metadata = field.metadata().clone();
         let children = match field.data_type() {
             DataType::Struct(children) => children
                 .iter()
                 .map(|f| Self::try_from(f.as_ref()))
                 .collect::<Result<_>>()?,
             DataType::List(item) => vec![Self::try_from(item.as_ref())?],
             DataType::LargeList(item) => vec![Self::try_from(item.as_ref())?],
+            DataType::Map(entries, keys_sorted) => {
+                // TODO: We only support keys_sorted=true for now,
+                //  because converting a rust arrow map field to the python arrow field will
+                //  lose the keys_sorted property.
+                if *keys_sorted {
```

Comment:
@Xuanwo I tried to restore the `keys_sorted` parameter in the Map type, which worked fine in Rust, but the parameter was lost once it converted to Python Arrow Schema using `PyArrowType`. I think this might be a bug with the Arrow code itself. Therefore, I want to only support Map type with `keys_sorted=false` in this PR, which is also the default value of Arrow MapType, and continue to investigate this issue. I will consider supporting Map type with `keys_sorted=true` in a future PR. But to be honest, this parameter does not actually sort the Map in Arrow; it is only used as metadata, so I don't think it will make a significant difference to users.

Reply:
Got it, thank you for working on this